classes/local/lib.php

```diff
@@ -146,4 +147,51 @@ public static function cm_info_view(cm_info $cminfo) {
         $renderer = new renderer($cms);
         $cminfo->set_content($renderer->get_html());
     }
+
+    /**
+     * Serves file
+     *
+     * @param stdClass $course
+     * @param stdClass $cm
+     * @param context $context
+     * @param string $filearea
+     * @param array $args
+     * @param bool $forcedownload
+     * @param array $options additional options affecting the file serving
+     * @return bool|null false if file not found, does not return anything if found - just send the file
+     */
+    public static function pluginfile($course, $cm, $context, $filearea, $args, $forcedownload, $options = []) {
+
+        // Make sure the filearea is one of those used by the plugin.
+        if ($filearea !== 'cms_type_images') {
+            return false;
+        }
+
+        require_login($course, true, $cm);
```

Comment:
I'd like to remove all the login and capability checks and serve these files with long expires headers so varnish caches them

Reply:
done